infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackService.kt

```diff
@@ -416,12 +416,28 @@ class LocationTrackService(
         return lines.map { line -> line to alignments.getValue(line.getAlignmentVersionOrThrow()) }
     }
 
+    fun sortDuplicatesByTrackAddress(
+        publicationState: PublicationState,
+        trackNumberId: IntId<TrackLayoutTrackNumber>,
+        duplicates: List<LocationTrackDuplicate>): List<LocationTrackDuplicate> {
+        val geocodingContext = geocodingService.getGeocodingContext(publicationState, trackNumberId)
+            ?: throw Exception("Failed to create geocoding context for track number $trackNumberId")
+        return duplicates.sortedBy { duplicate ->
+            val address = duplicate.duplicateStatus.startPoint?.let { start ->
+                geocodingContext.getAddress(start)?.first
```

Comment:
Huomiona ylempään, täällä hyväksytään null tulos ilman mitään heittoja. Joten tuosta voisi päätellä että null on OK. Jos näin, niin koko homma kannattaa tehdä ilman heittelyä muodossa:
```
return geocodingService.getGeocodingContext(publicationState, trackNumberId)?.let { ctx ->
  // Tänne tuo duplicates.sortedby jne
}
```